src/dataAccess/posts.ts

```diff
@@ -1,17 +1,30 @@
 import type { Post, PostResponse } from "@/types";
+import { BASE_API_URL } from "@/utils/constans";
 
-export const getPosts = async (url: string) => {
-  const res = await fetch(url);
-  if (!res.ok) {
-    throw new Error("Could not fetch the data for that resource");
-  }
-
-  const posts = await res.json() as PostResponse[];
+export const getPosts = async (
+  perPage: number,
+): Promise<[Post[] | null, Error | null]> => {
+  try {
+    const res = await fetch(`${BASE_API_URL}/posts?per_page=${perPage}`);
+    if (!res.ok) {
+      throw new Error("Could not fetch the data for that resource");
+    }
 
-  return posts.map((post) => ({
-    id: post.id,
-    title: post.title.rendered,
-    content: post.content.rendered,
-  }) as Post);
-}
+    const posts = (await res.json()) as PostResponse[];
+    const mappedPosts: Post[] = posts.map((post) => ({
+      id: post.id,
+      date: post.date,
+      title: post.title.rendered,
+      content: post.content.rendered,
+      excerpt: post.excerpt.rendered,
+      featured_media: post.featured_media,
+    }));
 
+    return [mappedPosts, null];
```

Comment:
This should be abstracted away into reusable format to return errors as values.
HTTP Client could be handling that or some middleware